contrib/IECoreUSD/src/IECoreUSD/USDScene.cpp

```diff
@@ -803,11 +823,25 @@ void USDScene::writeAttribute( const SceneInterface::Name &name, const Object *a
 	{
 		if( const Data *data = runTimeCast<const Data>( attribute ) )
 		{
-			pxr::UsdAttribute attribute = m_location->prim.CreateAttribute(
-				pxr::TfToken( name.string() ),
-				DataAlgo::valueTypeName( data )
-			);
-			attribute.Set( DataAlgo::toUSD( data ), time );
+			if ( boost::starts_with( name.string(), AttributeAlgo::userPrefix() ) || boost::starts_with( name.string(), AttributeAlgo::renderPrefix() ) )
```

Comment:
This is the reason I suggested the `bool &isPrimVarInUSD` output for `AttributeAlgo::toUSD()`. My intention was to encapsulate the information about prefixes within `AttributeAlgo`, rather than repeat it here. This gives us a path forwards for transforming `ai:*` attributes without needing any changes to USDScene itself.

Reply:
Sorry I didn't noticed how the `&isPrimVarInUSD` was not a const reference and thus what you had in mind regarding this.

I will answer your comment below because it applies to both of your last comments.